### ISSUE TITLE
ServiceStack.Text failing silently deserializing invalidly formatted (?) decimal value.

### ISSUE THREAD

**Issue by Martaver** [ServiceStack/Issues#480]
Hi, two parts to this... 

First is that the Json deserializer for `.FromJson<T>()` seems to fail silently in this example:

   ```
public class MyOrderBook
{
	public decimal[][] Bids { get; set; }
}

[Test]
public void ParseReturnOrderBookResponse()
{			
	var json = @"{""bids"": [[""0.01985141"", 7.67e-6]] }";			
	var response = json.FromJson<MyOrderBook>();
	Assert.That(response.Bids, Is.Not.Null);
}
```

Compare to here, where the example surfaces a `System.FormatException`:

```
var json = @"[[""0.01985141"", 7.67e-6]]";			
var response = json.FromJson<decimal[][]>();
Assert.That(response, Is.Not.Null);
```

I'm not sure if this is intended behavior, but if it is, what's the reasoning? I'm rather uncomfortable with formatting problems being swallowed behind the scenes.

Second part is that I think the Json deserializer should actually handle the '7.67e-6' in the first place, since it's a valid JSON value. In [this](http://stackoverflow.com/questions/18893159/how-to-parse-json-decimals-correctly-using-servicestack-jsonserializer) StackOverflow question, he points out that parsing the same string as a float seems to work. Could the JSON deserializer be smart enough to handle this for us?

**Comment by Martaver** [comment]
Some extra info... this is on Framework 4.5..

Also, I'm using this as a workaround, but should this be considered as the default deserializer fn?

```
JsConfig<decimal>.DeSerializeFn = s => decimal.Parse(s, NumberStyles.Float, CultureInfo.InvariantCulture);
```

**Comment by mythz** [comment]
ok changed to parse decimals as floats in this commit: https://github.com/ServiceStack/ServiceStack.Text/commit/c6ec2061743a248d1a20cd7163243a39634f9926

The change to use float style when parsing decimals is available in v4.5.1 that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet).

By default SS.Text tries to be resilient and silently deserialize as much as it can, you can change this behavior to throw on deserialization exception with:

```csharp
JsConfig.ThrowOnDeserializationError = true;
```

**Comment by Martaver** [comment]
Great! Thanks mate!
